sky/utils/kubernetes/generate_static_kubeconfig.sh

```diff
@@ -1,26 +1,38 @@
 #!/bin/bash
 # This script creates a new k8s Service Account and generates a kubeconfig with
-# its credentials. This Service Account has all the necessary permissions for
+# its credentials. This Service Account has the minimal permissions necessary for
 # SkyPilot. The kubeconfig is written in the current directory.
 #
-# You must configure your local kubectl to point to the right k8s cluster and
-# have admin-level access.
+# Before running this script, you must configure your local kubectl to point to
+# the right k8s cluster and have admin-level access.
 #
-# Note: all of the k8s resources are created in namespace "skypilot". If you
-# delete any of these objects, SkyPilot will stop working.
+# By default, this script will create a service account "sky-sa" in "default"
+# namespace. If you want to use a different namespace or service account name:
 #
-# You can override the default namespace "skypilot" using the
-# SKYPILOT_NAMESPACE environment variable.
-# You can override the default service account name "skypilot-sa" using the
-# SKYPILOT_SA_NAME environment variable.
+#   * Specify SKYPILOT_NAMESPACE env var to override the default namespace
+#   * Specify SKYPILOT_SA_NAME env var to override the default service account name
+#   * Specify SKIP_SA_CREATION=1 to skip creating the service account and use an existing one
+#
+# Usage:
+#   # Create "sky-sa" service account with minimal permissions in "default" namespace and generate kubeconfig
+#   $ ./generate_static_kubeconfig.sh
+#
+#   # Create "my-sa" account with minimal permissions in "my-namespace" namespace and generate kubeconfig
+#   $ SKYPILOT_SA_NAME=my-sa SKYPILOT_NAMESPACE=my-namespace ./generate_static_kubeconfig.sh
+#
+#   # Use an existing service account "my-sa" in "my-namespace" namespace and generate kubeconfig
+#   $ SKIP_SA_CREATION=1 SKYPILOT_SA_NAME=my-sa SKYPILOT_NAMESPACE=my-namespace ./generate_static_kubeconfig.sh
```

Comment:
Should we include the usage of this usage in the doc as well? So as to make it easier for the user to create the service account?

The doc for minimal permission above could serve as a deep dive into what this shell script do.

Reply:
absolutely - I'm working on it in a separate for revamping k8s cluster admin docs :)